pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1955,10 +1955,11 @@ public boolean getBoolean(int columnIndex) throws SQLException {
     }
 
     if (isBinary(columnIndex)) {
-      return BooleanTypeUtil.castToBoolean(readDoubleValue(this_row[col], fields[col].getOID(), "boolean"));
+      return BooleanTypeUtil.castToBoolean(readDoubleValue(this_row[col], fields[col].getOID(), "boolean"),
+          Oid.toString(fields[col].getOID()));
```

Comment:
The implementation of `Oid.toString(int)` does not look super fast.
Might be type is not necessary here? only value could be enough: `Cannot cast to boolean: "2"`